### ISSUE TITLE
Support import by alias or absolute paths

### ISSUE THREAD

**Issue by colinaaa** [UniqueStudio/UniqueRecruitmentBackend#16]
Since Node does not support absolute `require()` and typescript does not support rewrite absolute import or alias import to relative import which Node `require()` can recognize, we have to import module like this:

```ts
import { GENDERS, GRADES, GROUPS_, RANKS } from '../../config/consts';
import { CandidateRepo, RecruitmentRepo } from '../../database/model';
import { copyFile } from '../../utils/copyFile';
import { errorRes } from '../../utils/errorRes';
import { logger } from '../../utils/logger';
import sendEmail from '../../utils/sendEmail';
```

But VSCode's auto-import for typescript always results as absolute import, which Node will complain about.

eg.
https://github.com/UniqueStudio/UniqueRecruitmentBackend/blob/21823a6489d3143065bd16adf9f07ce54474175e/src/utils/shortenURL.ts#L2

Maybe we could use [tsc-alias](https://github.com/justkey007/tsc-alias) or [module-alias](https://github.com/ilearnio/module-alias) to prevent things like this happening again.
It can also change the horrible relative paths more human-readable.

**Comment by colinaaa** [comment]
I had a little script in my personal project which use [tsc-alias](https://github.com/justkey007/tsc-alias) to solve similar problems.

```ts
// scripts/resolveAlias.ts
import { resolve } from 'path';

// eslint-disable-next-line import/no-extraneous-dependencies
import { replaceTscAliasPaths } from 'tsc-alias';

const rootPath = resolve(__dirname, '../');

replaceTscAliasPaths({
  configFile: resolve(rootPath, 'tsconfig.json'),
});
```

run `ts-node scripts/resolveAlias.ts` after `tsc` can resolve all alias and absolute import.

**Comment by ManiaciaChao** [comment]
写个 postbuild 的 script 挺好的

**Comment by ManiaciaChao** [comment]
> 写个 postbuild 的 script 挺好的

但是直接 ts-node 感觉会翻车

**Comment by colinaaa** [comment]
> 但是直接 ts-node 感觉会翻车

确实，直接ts-node还需要[tsconfig-paths](https://github.com/dividab/tsconfig-paths)
那不如直接都用它算了

**Comment by colinaaa** [comment]
implemented in 62f9380, closed
